### ISSUE TITLE
Add torch.narrow_scatter operator to complete the *_copy/*_scatter operator family.

### ISSUE THREAD

**Issue by Sanskaaar1** [pytorch/pytorch#177372]
### The feature, motivation and pitch

PyTorch currently has `narrow_copy` but lacks the corresponding `narrow_scatter` operator, breaking the pattern where every `*_copy` operation has a matching `*_scatter` operation:
- `slice_copy` / `slice_scatter` - exists
- `select_copy` / `select_scatter` - exists
- `narrow_copy` / `narrow_scatter` - missing

Without `narrow_scatter`, the functorch functionalization layer can't properly handle mutations through narrow views. Right now users have to use mutation-based workarounds that don't work with functorch:

```python
# Current workaround (not functional, incompatible with functorch)
result = input.clone()
result.narrow(dim, start, length).copy_(src)
```

I'd like to propose adding `torch.narrow_scatter(input, src, dim, start, length) -> Tensor` which scatters values from `src` into a narrow region of `input`:

```python
import torch

# Basic usage
x = torch.zeros(10)
src = torch.ones(5) * 2
result = torch.narrow_scatter(x, src, dim=0, start=2, length=5)
# tensor([0., 0., 2., 2., 2., 2., 2., 0., 0., 0.])

# Autograd support
x = torch.randn(10, 10, requires_grad=True)
src = torch.randn(5, 10, requires_grad=True)
result = torch.narrow_scatter(x, src, 0, 2, 5)
result.sum().backward()  # gradients computed correctly

# torch.compile support
@torch.compile
def fn(x, src):
    return torch.narrow_scatter(x, src, 0, 2, 5)
```

The implementation can delegate to `slice_scatter` since narrow is just slice with step=1:

```cpp
at::Tensor narrow_scatter_symint(..., SymInt start, SymInt length) {
  c10::SymInt end = start + length;
  return at::slice_scatter_symint(self, src, dim, start, end, 1);
}
```

This keeps the implementation minimal and reuses all existing `slice_scatter` optimizations.

[Related PR #175609](https://github.com/pytorch/pytorch/pull/175609/changes)

### Alternatives

Users can work around this with `slice_scatter` directly:
```python
result = torch.slice_scatter(input, src, dim, start, start+length, 1)
```

However, this is less intuitive (requires computing `end = start + length`) and inconsistent with the `narrow_copy` naming convention.

The mutation-based workaround also exists but doesn't work with functorch:
```python
result = input.clone()
result.narrow(dim, start, length).copy_(src)
```

### Additional context

[Related PR #175609](https://github.com/pytorch/pytorch/pull/175609/changes)

cc @pragupta @bdhirsh @ezyang @chauhang @penguinwu @Chillee @samdow @kshitij12345 @bobrenjc93 @aorenste
